### ISSUE TITLE
Forward references don't seem to work for Generic type variables

### ISSUE THREAD

**Issue by aspin** [facebook/pyre-check#153]
Here's what works:

```python
from typing import TypeVar, Generic


class Attribute:
    name: str
    model: "Model"

    def __init__(self, name: str):
        self.name = name


T = TypeVar("T", bound=Attribute)


class NumberedAttribute(Attribute):
    number: int = 1


class Model(Generic[T]):
    attribute: T

    def __init__(self, attribute: T):
        self.attribute = attribute


class NumberedModel(Model[NumberedAttribute]):

    def do_stuff(self) -> int:
        return self.attribute.number
```

But if you change `Model[NumberedAttribute]` to `Model["NumberedAttribute"]` (necessary if you move `NumberedAttribute` to a different file and have a circular reference), then we get this error:

```
 ƛ Found 1 type error!
src/ptest/main.py:25:15 Undefined attribute [16]: `Variable[T (bound to Attribute)]` has no attribute `number`.
```

**Comment by mrkmndz** [comment]
Thanks for the report, we may need to add support for parsing string annotations in inheritance declarations.

**Comment by aspin** [comment]
I actually found a workaround.

```python
NumberedAttributeType = TypeVar("NumberedAttributeType", bound="NumberedAttribute")


class NumberedModel(Model[NumberedAttributeType]):
    # ...
```

Not idea, but works in the meantime.

**Comment by mrkmndz** [comment]
FYI that has different semantics than your previous example, as that makes NumberedModel a generic class as well.

**Comment by aspin** [comment]
Just to double check, that means this is possible right?:

```python
class SuperNumberedAttribute(NumberedAttribute):
    is_super: bool = True

class SuperNumberedModel(NumberedModel[SuperNumberedAttribute]):
    pass
```

Not really desired, but I suppose it gets builds passing and autocomplete happy for now.

**Comment by mrkmndz** [comment]
Fixed by fe8f33807d7089d3d70383d36fee5801da4f6824
